lib/lists.nix

```diff
@@ -687,6 +687,47 @@ rec {
    */
   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
 
+  /* Check if list contains only unique elements.
+
+     Type: allUnique :: [a] -> bool
+
+     Example:
+       allUnique [ 3 2 3 4 ]
+       => false
+       allUnique [ 3 2 4 1 ]
+       => true
+   */
+  allUnique = lst: (builtins.length (unique lst) == builtins.length lst);
```

Comment:
There's a potential laziness optimisation here: Exit early as soon as duplicate elements are found

Reply:
I get the idea, but I am not much experienced...
Not sure if or "how to do" this lazy...
```
allUnique = list: (foldl' (acc: e: { r = !(elem e acc.seen) && acc.r; seen = acc.seen ++ [ e ]; } ) {seen = []; r= true;} list).r;
```
will continue tomorrow.

Reply:
I think let's leave the code as it is currently. It's unlikely for this to matter much :)